packages/ui/src/components/ChatMessage/ChatMessageActions/MessageActions.tsx

```diff
@@ -315,3 +221,80 @@ export async function handleAction({
   await Haptics.notificationAsync(Haptics.NotificationFeedbackType.Success);
   dismiss();
 }
+
+/**
+ * Extra information about how to display the action. This can change based on
+ * the UI context - e.g. the label for `startThread` changes based on channel
+ * type.
+ */
+export function useDisplaySpecForChannelActionId(
+  id: ChannelAction.Id,
+  {
+    post,
+    channel,
+  }: {
+    post: db.Post;
+    channel: db.Channel;
+  }
+): {
+  label: string;
+} {
+  const isMuted = logic.isMuted(post.volumeSettings?.level, 'thread');
+  const postTerm = useMemo(() => {
+    return ['dm', 'groupDm', 'chat'].includes(channel?.type)
+      ? 'message'
+      : 'post';
+  }, [channel?.type]);
+
+  return useMemo(() => {
+    switch (id) {
+      case 'copyRef':
+        return {
+          label:
+            postTerm === 'message'
+              ? 'Copy link to message'
+              : 'Copy link to post',
+        };
+
+      case 'copyText':
+        return { label: 'Copy message text' };
+
+      case 'delete':
+        return {
+          label: postTerm === 'message' ? 'Delete message' : 'Delete post',
```

Comment:
Any reason to not just use a template string here?

Reply:
if I wanted to change the label text, I'd search the codebase for `Delete post` – if this was `Delete ${type === 'post' ? 'post' : 'message'}` or similar, I wouldn't be able to easily find this string